### ISSUE TITLE
Fix loses generic

### ISSUE THREAD

**Issue by OliverJAsh** [JamieMason/eslint-plugin-prefer-arrow-functions#10]
## Description

Given

```ts
function identity<T>(t: T): T { return t; }
```

when we run the fix we get this:

```ts
const identity = (t: T) => t;
```

I expected to get this:

```ts
const identity = <T>(t: T) => t;
```

**Comment by JamieMason** [comment]
Thanks @OliverJAsh, I've not been doing a lot of OSS lately but I will get to this.

**Comment by JamieMason** [comment]
Released in `3.1.3`, I've only added your examples above as test cases so let me know if you run into any problems.
